src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -25,42 +28,12 @@ public CSharpPullMemberUpCodeRefactoringProvider() : base(null)
         {
         }
 
-        protected override bool IsSelectionValid(TextSpan span, SyntaxNode selectedNode)
+        protected override async Task<SyntaxNode> GetSelectedNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            var identifier = GetIdentifier(selectedNode);
-            if (identifier == default)
-            {
-                return false;
-            }
-            else if (identifier.FullSpan.Contains(span) && span.Contains(identifier.Span))
-            {
-                // Selection lies within the identifier's span
-                return true;
-            }
-            else if (identifier.Span.Contains(span) && span.IsEmpty)
-            {
-                // Cursor stands on the identifier
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
-
-        private SyntaxToken GetIdentifier(SyntaxNode selectedNode)
-        {
-            switch (selectedNode)
-            {
-                case MemberDeclarationSyntax memberDeclarationSyntax:
-                    // Nested type is checked in before this method is called.
-                    return memberDeclarationSyntax.GetNameToken();
-                case VariableDeclaratorSyntax variableDeclaratorSyntax:
-                    // It handles multiple fields or events declared in one line
-                    return variableDeclaratorSyntax.Identifier;
-                default:
-                    return default;
-            }
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
+            return await refactoringHelperService.TryGetSelectedNodeAsync(
+                document, span, Functions<SyntaxNode>.Identity,
+                (n) => n is MemberDeclarationSyntax || n is VariableDeclaratorSyntax, cancellationToken).ConfigureAwait(false);
```

Comment:
1. no need for parens in `(n)`
2. having to pass ing `Functions<SyntaxNode>.Identity` makes for a very strange piece of code to try to grok.

Reply:
1) (y)
2) True, will either add overload or add cached version on the service with more descriptive name.

Reply:
It'll get solved by having default extractor (as decided in other convo) -> no identity will be used.